File: frontend/src/app/models/Seed.ts
import { SeedType } from "./SeedType";

export interface Seed {
  id: string;
  seedType: string;
  displayName: string;
  scientificName: string;

  // Temperature
  minTemperature: number;
  maxTemperature: number;

  // Growth timeline - ADD THESE FIELDS
  daysToYoung: number;
  daysToMature: number;
  daysToReady: number;

  // Water & Soil
  waterRequirement: number;
  minSoilMoisture: number;
  allowedWaterDeficit: number;

  // Stress thresholds
  heatStressTemperature: number;
  heavyRainThreshold: number;
  maxWindTolerance: number;

  // Disease risk
  diseaseRiskMinTemp: number;
  diseaseRiskMaxTemp: number;
  diseaseRainThreshold: number;

  // Other
  seedCoefficient: number;
}

File: backend/farm-service/src/main/java/ase_pr_inso_01/farm_service/models/Seed.java
package ase_pr_inso_01.farm_service.models;

import ase_pr_inso_01.farm_service.models.enums.SeedType;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "seed")
@Getter
@Setter
public class Seed {
    @Id
    private String id;

    private SeedType seedType;
    private String displayName;
    private String scientificName;

    private Double minTemperature;
    private Double maxTemperature;
    private Double heatStressTemperature;

    private Double waterRequirement;
    private Double heavyRainThreshold;
    private Double minSoilMoisture;
    private Double allowedWaterDeficit;
    private Double seedCoefficient;

    private Integer daysToYoung;
    private Integer daysToMature;
    private Integer daysToReady;
    private Double diseaseRiskMinTemp;
    private Double diseaseRiskMaxTemp;
    private Double diseaseRainThreshold;

    private Double maxWindTolerance;
}